File: rockgame.py
import random
print(" Welcome to rock , paper, scissors game:")
print("0=ROCK, 1=PAPER, 2=SCISSOR")
choices = [0, 1, 2]
userchoice = input("enter userchoice(0 to 2):")
computerchoice = random.choice(choices)
if (userchoice == computerchoice):
    print("It is tie!")

elif((userchoice == 0 and computerchoice == 2) or (userchoice == 1 and computerchoice == 0) or (userchoice == 2 and computerchoice == 1)):
    print("you win!")
elif((userchoice == 2 and computerchoice == 0) or (userchoice == 0 and computerchoice == 1) or (userchoice == 1 and computerchoice == 2)):
    print("you loss!!")

print("userchoice = ", userchoice)
print("computerchoice = ", computerchoice)

File: rockgame.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>


int getComputerChoice() {
    return rand() % 3; 
}


void determineWinner(int userChoice, int computerChoice) {
    if (userChoice == computerChoice) {
        printf("It's a tie!\n");
    } else if ((userChoice == 0 && computerChoice == 2) || 
               (userChoice == 1 && computerChoice == 0) || 
               (userChoice == 2 && computerChoice == 1)) {
        printf("You win!\n");
    } else {
        printf("You lose!\n");
    }
}

int main() {
    int userChoice, computerChoice;
    
    
    srand(time(0));

    printf("Welcome to Rock, Paper, Scissors!\n");
    scanf("%d", &userChoice);

    
    if (userChoice < 0 || userChoice > 2) {
        printf("Invalid choice! Please choose a number between 0 and 2.\n");
        return 1;
    }

    
    computerChoice = getComputerChoice();

    
    const char* choices[] = {"Rock", "Paper", "Scissors"};
    printf("You chose: %s\n", choices[userChoice]);
    printf("Computer chose: %s\n", choices[computerChoice]);

    
    determineWinner(userChoice, computerChoice);

    return 0;
}

